### PR TITLE
add doc for typescript md

### PR DESCRIPTION
**Pull request by qiaozha**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by qiaozha** [comment]
@josefree  @akning-ms  This PR is try to add document for typescript sdk. Can you help review it? Thanks

File: documentation/code-gen/configure-typescript-sdk.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+### Basic Information
+Configure package title/description/tag.
+~~~~
+// file: readme.md
+
+``` yaml
+title: xxxxConfigurationClient
+description: xxxx Configuration Client
+openapi-type: arm
+tag: package-xxxx-xx-xx
```
Comment (akning-ms) [inline documentation/code-gen/configure-typescript-sdk.md:16]: Should we add some name rule for tag to make it consistent? and give some guide for when we will use this tag

Comment (qiaozha) [reply documentation/code-gen/configure-typescript-sdk.md:16]: I am fine to add some tag name convention for the consistency consideration. But should we discuss what kind of convention is needed in a more general way? For example. what the tag name would be in the scenarios of multi-package, in the scenarios of preview/stable version, in the scenarios of multi-api  etc ? Also I noticed that we use YYYY-mm to distinguish different versions previously, what if the service team has two api-versions such as 2020-05-01 and 2020-05-16 in one month?

Comment (akning-ms) [reply documentation/code-gen/configure-typescript-sdk.md:16]: You can refer to changlong's PR for python

Comment (akning-ms) [reply documentation/code-gen/configure-typescript-sdk.md:16]: PR #10123

Comment (qiaozha) [reply documentation/code-gen/configure-typescript-sdk.md:16]: Okay. I will change it

Comment (qiaozha) [reply documentation/code-gen/configure-typescript-sdk.md:16]: @akning-ms I have update the tag description like #10123 . Could you please review again? Thanks

**Review by akning-ms — APPROVED** [review]

File: documentation/code-gen/configure-typescript-sdk.md [hunk 2]
```diff
@@ -0,0 +1,235 @@
+# Readme Configuration Guide for Typescript SDK
```
Comment (nickzhums) [inline documentation/code-gen/configure-typescript-sdk.md:1]: I would elaborate this name a bit for clarification

Azure SDK for Javascript (Node.js and browser) = TypeScript SDK

We may want to be explicit about this to not confuse people with the deprecated node.js SDK repo

**Review by nickzhums — APPROVED** [review]

**Pull request merged**